test/standardizedToken.js

```diff
@@ -0,0 +1,304 @@
+
+describe('StandardizedToken', () => {
+    let standardizedToken, standardizedTokenDeployer;
+
+    const name = 'tokenName';
+    const symbol = 'tokenSymbol';
+    const decimals = 18;
+    const mintAmount = 123;
+
+    let token;
+    let tokenProxy;
+
+    let owner, user;
+
+    before(async () => {
+        const wallets = await ethers.getSigners();
+        owner = wallets[0];
+        user = wallets[1];
+    });
+
+    beforeEach(async () => {
+        standardizedToken = await deployContract(owner, 'StandardizedToken');
+        standardizedTokenDeployer = await deployContract(owner, 'StandardizedTokenDeployer', [standardizedToken.address]);
+
+        const salt = getRandomBytes32();
+
+        const tokenAddress = await standardizedTokenDeployer.deployedAddress(salt);
+
+        token = new Contract(tokenAddress, StandardizedToken.abi, owner);
+        tokenProxy = new Contract(tokenAddress, StandardizedTokenProxy.abi, owner);
+
+        await standardizedTokenDeployer.deployStandardizedToken(
+            salt,
+            owner.address,
+            owner.address,
+            name,
+            symbol,
+            decimals,
+            mintAmount,
+            owner.address,
+        );
+    });
+
+    describe('Standardized Token Proxy', () => {
+        it('should revert if standardized token implementation is invalid', async () => {
+            const invalidStandardizedToken = await deployContract(owner, 'InvalidStandardizedToken');
+            standardizedTokenDeployer = await deployContract(owner, 'StandardizedTokenDeployer', [invalidStandardizedToken.address]);
+
+            const salt = getRandomBytes32();
+
+            await expect(
+                standardizedTokenDeployer.deployStandardizedToken(
+                    salt,
+                    owner.address,
+                    owner.address,
+                    name,
+                    symbol,
+                    decimals,
+                    mintAmount,
+                    owner.address,
+                    getGasOptions(),
+                ),
+            ).to.be.reverted;
+        });
+
+        it('should return the correct contract ID', async () => {
+            const contractID = await token.contractId();
+            const hash = keccak256(toUtf8Bytes('standardized-token'));
+            expect(contractID).to.equal(hash);
+        });
+    });
+
+    describe('Standardized Token', () => {
+        it('revert on setup if not called by the proxy', async () => {
+            const implementationAddress = await tokenProxy.implementation();
+            const implementation = new Contract(implementationAddress, StandardizedToken.abi, owner);
+
+            const params = '0x';
+            await expectRevert((gasOptions) => implementation.setup(params, gasOptions), token, 'NotProxy');
+        });
+    });
+
+    describe('ERC20 Basics', () => {
```

Comment:
ERC20 and ERC20 Permit tests should be within their ERC20.js and ERC20Permit.js files. The contract specific test should cover code defined in that contract

Reply:
I separated them into different files